src/main/java/unibook/logic/commands/EditCommand.java

```diff
@@ -44,41 +54,81 @@
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_OPTION + "person "
+            + PREFIX_PHONE + "91234567 "
+            + PREFIX_EMAIL + "johndoe@example.com";
 
     public static final String MODULE_MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the module identified "
-        + "by the index number used in the displayed module list. "
-        + "Existing values will be overwritten by the input values.\n"
-        + "Parameters: INDEX (must be a positive integer) "
-        + PREFIX_OPTION + "OPTION "
-        + PREFIX_NAME + "NAME "
-        + "[" + PREFIX_MODULE + "MODULECODE] "
-        + "[" + PREFIX_NEWMOD + "NEWMODULECODE] \n "
-        + "Example: " + COMMAND_WORD + " 1 "
-        + PREFIX_OPTION + "module "
-        + PREFIX_NAME + "Software Engineering "
-        + PREFIX_MODULE + "CS2103T";
+            + "by the index number used in the displayed module list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_OPTION + "OPTION "
+            + PREFIX_NAME + "NAME "
+            + "[" + PREFIX_MODULE + "MODULECODE] "
+            + "[" + PREFIX_NEWMOD + "NEWMODULECODE] \n "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_OPTION + "module "
+            + PREFIX_NAME + "Software Engineering "
+            + PREFIX_MODULE + "CS2103T";
+
+    public static final String GROUP_MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the group identified "
+            + "by the index number used in the displayed group list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_OPTION + "OPTION "
+            + PREFIX_MODULE + "MODULECODE "
+            + "[" + PREFIX_GROUP + "GROUPNAME] "
+            + "[" + PREFIX_MEETINGTIME + "MEETINGTIME] \n "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_OPTION + "group "
+            + PREFIX_NAME + "CS2103 Team1 "
+            + PREFIX_MEETINGTIME + "12-12-2022 1645";
+
+    public static final String KEYEVENT_MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the key event identified"
+            + " by the index number used in the displayed module list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (of the module must be a positive integer) "
+            + PREFIX_OPTION + "OPTION "
+            + "[" + PREFIX_KEYEVENT + "KEYEVENT] "
+            + "[" + PREFIX_DATETIME + "DATETIME] \n "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_OPTION + "keyevent "
+            + PREFIX_KEYEVENT + "1 "
+            + PREFIX_TYPE + "exam "
+            + PREFIX_MEETINGTIME + "12-12-2022 1645";
 
```

Comment:
One way to handle the lack of `option` is to have an enum type here, similar to what I have pasted in the screenshot.
![image](https://user-images.githubusercontent.com/83899964/160528338-a265db1d-e5c5-4d27-a1a3-934bce1ee385.png)


Then when you create a command in `EditCommandParser`, you can pass in the respective type (removes the need for all the boolean checks, etc cos you know that the respective fields you need will not be null when you create a command with the specific enum. To handle different behaviour across different views, the `execute` method has booleans to show which group list is showing.

![image](https://user-images.githubusercontent.com/83899964/160528518-f89e082e-e189-4a1b-9bf2-fd43b63bd6ef.png)


This is an example - see case `module` which is the same command that behaves differently on differnet pages.